library/ssl_tls.c

```diff
@@ -3142,6 +3142,29 @@ void mbedtls_ssl_init( mbedtls_ssl_context *ssl )
     memset( ssl, 0, sizeof( mbedtls_ssl_context ) );
 }
 
+static int ssl_conf_check(const mbedtls_ssl_context *ssl)
```

Comment:
I think `ssl_conf_check()` should return an SSL error code instead of `0,1`. How about something like this:
```c
static int ssl_conf_check( mbedtls_ssl_context const *ssl )
{
    if( !mbedtls_ssl_conf_is_tls12_only( ssl->conf ) &&
        !mbedtls_ssl_conf_is_tls11_only( ssl->conf ) )
    {
        MBEDTLS_SSL_DEBUG_MSG( 1, "Hybrid TLS 1.2 + TLS 1.3 configurations are not yet supported" );
        return( MBEDTLS_ERR_SSL_FEATURE_UNAVAILABLE );
    }

    /* Space for further checks in the future */

    return( 0 );
}

Reply:
And then later
```c
if( ( ret = ssl_conf_check( ssl ) ) != 0 ) 
    return( ret );
````

Reply:
It should still return bad config when it is not hybrid, tls1_2 only and tls1_3 only.

Reply:
I think `FEATURE_UNAVAILABLE` is more fitting, because the config isn't inherently bad, but we just haven't implemented support for dual 1.2-1.3 yet.

Reply:
Yes, that's right!